bundles/org.openhab.binding.somfymylink/src/main/java/org/openhab/binding/somfymylink/internal/handler/SomfyMyLinkBridgeHandler.java

```diff
@@ -0,0 +1,447 @@
+                    return;
+                }
+
+                if (CHANNEL_SCENES.equals(channelUID.getId()) && command instanceof StringType) {
+                    Integer sceneId = Integer.decode(command.toString());
+                    commandScene(sceneId);
+                }
+            }
+        } catch (SomfyMyLinkException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing mylink");
+        config = getThing().getConfiguration().as(SomfyMyLinkConfiguration.class);
+
+        if (validConfiguration(config)) {
+            this.discoveryServiceRegistration = FrameworkUtil.getBundle(SomfyMyLinkBridgeHandler.class)
+                    .getBundleContext().registerService(DiscoveryService.class, this.discovery, null);
+            this.discovery.activate(null);
+
+            // kick off the bridge connection process
+            this.scheduler.execute(this::connect);
+        }
+    }
+
+    private boolean validConfiguration(@Nullable SomfyMyLinkConfiguration config) {
+        if (config == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "mylink configuration missing");
+            return false;
+        }
+
+        if (StringUtils.isEmpty(config.ipAddress)) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "mylink address not specified");
+            return false;
+        }
+
+        return true;
+    }
+
+    private void connect() {
+        try {
+            if (config == null) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "mylink config not specified");
+                return;
+            }
+
+            if (StringUtils.isEmpty(config.ipAddress) || StringUtils.isEmpty(config.systemId)) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "mylink config not specified");
+                return;
+            }
+
+            // start the keepalive process
+            ensureKeepAlive();
+
+            logger.debug("Connecting to mylink at {}", config.ipAddress);
+
+            // send a ping
+            sendPing();
+
+            logger.debug("Connected to mylink at {}", config.ipAddress);
+
+            updateStatus(ThingStatus.ONLINE);
+        } catch (SomfyMyLinkException e) {
+            logger.debug("Problem connecting to mylink, bridge OFFLINE");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+    }
+
+    private void ensureKeepAlive() {
+        if (heartbeat == null) {
+            logger.debug("Starting keepalive job in {} min, every {} min", HEARTBEAT_MINUTES, HEARTBEAT_MINUTES);
+            heartbeat = this.scheduler.scheduleWithFixedDelay(this::sendKeepAlive, 1, 1, TimeUnit.MINUTES);
+        }
+    }
```

Comment:
should be synchronized to prevent multiple `heartbeat` instances from being created at once.

Reply:
done